src/components/ChatApp/ChatApp.react.js

```diff
@@ -11,18 +10,44 @@ import { createMessagePairArray } from '../../utils/formatMessage';
 class ChatApp extends Component {
   static propTypes = {
     actions: PropTypes.object,
+    app: PropTypes.object,
   };
 
+  constructor(props) {
+    super(props);
+    const { uuid, email, userName, accessToken } = props.app;
+    this.state = {
+      uuid,
+      email,
+      userName,
+      accessToken,
+    };
+  }
+
   componentDidMount() {
+    this.getMessageHistory();
     document.title = 'SUSI.AI Chat - Open Source Artificial Intelligence';
-    // force an update if the URL changes
-    history.listen(() => this.forceUpdate());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { uuid, email, userName, accessToken } = nextProps.app;
+    if (uuid !== this.state.uuid) {
+      this.getMessageHistory();
+      this.setState({
```

Comment:
If this is done through redux, action can be dispatch and the props will change. As the props changes, the view will be re-rendered.

Reply:
Yup.. correct, good catch. This will just be needed to make sure that we only fetch new message when there is a change in uuid.

Reply:
`uuid` is not needed here. Incase user logs out, we need to call the action to fetch messages from there itself.

Reply:
This may cause unnecessary re-renders.